llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -549,24 +552,33 @@ void AIEPostRASchedStrategy::commitBlockSchedule(MachineBasicBlock *BB) {
   // Safety margin, swp epilogue
   InterBlock.emitInterBlockTop(BS);
 
-  MachineBasicBlock::iterator It = BB->begin();
-  const TargetInstrInfo *TII = getTII(BB);
-  for (auto &Region : BS.getRegions()) {
-    // Contrary to PRAS, the MachineScheduler does not automatically insert
-    // NOPs. That isn't a problem, since the callbacks to the HazardRecognizer
-    // were a bit flaky (e.g. when to call emitNoop vs advanceCycle).
-    // MachineScheduler just calls advanceCycle, and this is enough for us to
-    // insert NOPs because the sequence of Bundles gives us the full picture.
-    for (const AIE::MachineBundle &Bundle : Region.Bundles) {
-      if (Bundle.empty()) {
-        // Empty bundle means 1-cycle stall.
-        TII->insertNoop(*BB, It);
-        continue;
+  if (BS.isPipelined()) {
+    assert(BS.getRegions().size() == 1);
+    MachineBasicBlock::iterator It = BB->getFirstTerminator();
+    const bool Move = true;
+    InterBlock.emitBundles(BS.getRegions().front().Bundles, BB, It, Move);
+  } else {
+    MachineBasicBlock::iterator It = BB->begin();
+    const TargetInstrInfo *TII = getTII(BB);
+    for (auto &Region : BS.getRegions()) {
+      // Contrary to PRAS, the MachineScheduler does not automatically insert
+      // NOPs. That isn't a problem, since the callbacks to the HazardRecognizer
+      // were a bit flaky (e.g. when to call emitNoop vs advanceCycle).
+      // MachineScheduler just calls advanceCycle, and this is enough for us to
+      // insert NOPs because the sequence of Bundles gives us the full picture.
+      for (const AIE::MachineBundle &Bundle : Region.Bundles) {
+        if (Bundle.empty()) {
+          // Empty bundle means 1-cycle stall.
+          TII->insertNoop(*BB, It);
+          continue;
+        }
+        It = std::next(Bundle.getInstrs().back()->getIterator());
       }
-      It = std::next(Bundle.getInstrs().back()->getIterator());
+      AIEHazardRecognizer::applyBundles(Region.Bundles, BS.TheBlock);
     }
-    AIEHazardRecognizer::applyBundles(Region.Bundles, BS.TheBlock);
   }
+  // swp prologue
+  InterBlock.emitInterBlockBottom(BS);
```

Comment:
Similar comment here: I'm wondering why this is called `emitInterBlockBottom` if we insert things at the top of the block?